### ISSUE TITLE
Feature: Generalistic improvements after first release

### ISSUE THREAD

**Issue by ev1stensberg** [webpack/webpack-cli#87]
Lately been looking at the Polymer CLI and it looks really neat. Not only is it really well written, but it's elegantly looking. 

This is a sum of things I think we should fix once we got the time:

- Improve the visuals of the CLI
- Improve yargs and how we process arguments on our own
- Make a logger module for webpack instead of manually configuring this. ( Useful for both the CLI and webpack itself )

**Comment by alejandronanez** [comment]
I think I can grab something from this list, is all listed there still a thing? Any good beginner steps that I can follow?
Thanks!

**Comment by ev1stensberg** [comment]
I think the best thing to start on is E2E tests! You can look at yarn: https://github.com/yarnpkg/yarn/tree/master/end_to_end_tests on how they do that! 

To start dev:

1. Clone repo
2. npm link and remove the global webpack-cli installation if you have any
3. Happy hacking :)

**Comment by alejandronanez** [comment]
Sounds good, will work on E2E then!

**Comment by ev1stensberg** [comment]
Hit me up if you need help, available anytime 👍

**Comment by alejandronanez** [comment]
Hey @ev1stensberg, can you give me a quick overview about what is the purpose of e2e for `webpack-cli` it should be similar to E2E for an app but I'm still kinda lost tbh :/
Thanks for your help!

**Comment by ev1stensberg** [comment]
E2E is to test the CLI, similar to smoketests

**Comment by ev1stensberg** [comment]
@alejandronanez Have you started with E2E yet? I think we can simplify this a lot through doing `stdin` mocks with Jest for init and migrate, and watch for a successfully created `webpack.config.js` with init in Docker containers.

**Comment by ematipico** [comment]
As agreed with @ev1stensberg, I will take care of integrating Flow

**Comment by thescientist13** [comment]
@ev1stensberg 
What did you have in mind for this task?  (if it's still available)
> Make a logger module for webpack instead of manually configuring this. ( Useful for both the CLI and webpack itself )

**Comment by ev1stensberg** [comment]
I think that @bebraw linked to an issue that's open at the core repository. You could align with him to get info!

**Comment by thescientist13** [comment]
Thanks @ev1stensberg !

@bebraw 
Do you have any thoughts on direction I could review to get started on this?  Thanks in advance!

**Comment by bebraw** [comment]
@thescientist13 Not really. The logger module is something you should discuss on webpack core. Propose an API for that. The rest seem isolated to the CLI.

**Comment by dhruvdutt** [comment]
@ev1stensberg Any suggestions which one to start off with as a first time contributor?

**Comment by ev1stensberg** [comment]
End to End tests would be gold!

**Comment by dhruvdutt** [comment]
@ev1stensberg Can you please check Gitter? Need some quick help.

**Comment by rishabh3112** [comment]
I was researching for E2E (as labeled for beginners) part of this issue, where i explored end to end tests that yarn do [here](https://github.com/yarnpkg/yarn/tree/master/end_to_end_tests).
There I found that they use docker container to run tests, I understood what a container is but facing problems for it's setup can anybody help?
and also what sort of tests to run?

**Comment by rishabh3112** [comment]
@ev1stensberg ?

**Comment by ev1stensberg** [comment]
Hi @rishabh3112 . Run tests that make webpack configurations that are valid. ( Smoketests ). Have a look at `test/testBinCases` on how to setup that. Did I answer your question?

**Comment by rishabh3112** [comment]
Yes, and can I also know which all cases are left to be implemented @ev1stensberg

**Comment by ev1stensberg** [comment]
Awesome. Do you have to get started?

**Comment by rishabh3112** [comment]
I have cloned repo and installed packages and packages of individual packages in packages folder.
Have run tests present currently which all passed.
Learning jest and the structure of binCases present.
Want to know next step

**Comment by rishabh3112** [comment]
@ev1stensberg ?

**Comment by ev1stensberg** [comment]
Hi! Do you want to start with an easier issue ? This might be a hard one for complete beginners. I can give you a better one 🤗

**Comment by rishabh3112** [comment]
Yes you can give an easier one too but i also want my hands on this.😃

**Comment by ev1stensberg** [comment]
The first step would be to create an image or container of the latest version and then adding the tests

**Comment by rishabh3112** [comment]
Is it okay to ask doubts in future as i am a beginner in all these things. 😅

**Comment by ev1stensberg** [comment]
No worries @rishabh3112 , available to answer questions!

**Comment by rishabh3112** [comment]
@ev1stensberg will I have to setup docker on my computer or there is any option available to setup at any remote server ? my computer doesn't have enough resourses

**Comment by ev1stensberg** [comment]
I think you'll need to install it locally

**Comment by rishabh3112** [comment]
I am facing this issue when i run `docker run hello-world` (hello-world is sample container for checking installation)
`docker: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?.`
using Ubuntu 16.04.2 LTS
and docker version 18.03.1-ce
Needed Help @ev1stensberg

**Comment by ev1stensberg** [comment]
I think Google would be your best friend on these kinds of issues

**Comment by rishabh3112** [comment]
Nothing helped, i post here because i thought someone know here maybe.
Struck on for a day

**Comment by ev1stensberg** [comment]
No worries, take your time :) Google docker for beginners and read up on articles and such!

**Comment by rishabh3112** [comment]
Can't install docker on my system as I use Ubuntu using WSL. Installing docker on it linux doesn't work as docker's daemon can't run it's background tasks .Docker can be easily setup on WSL using docker for windows too but that requires Windows 10 Pro edition but I have Windows 10 Home. I tried using `rkt` but that too was of no help. So, I may not be able to help in this issue so can I have any other @ev1stensberg.

**Comment by rishabh3112** [comment]
Yes, I can work on documentations.
Should we move to #247 or open a new issue @ev1stensberg

**Comment by ev1stensberg** [comment]
Move to 247. Comment what you would like to work on :)

**Comment by AjayPoshak** [comment]
Is anything left here up for grabs?
I would like to contribute.

**Comment by rishabh3112** [comment]
Yes @AjayPoshak, You can get started with E2E tests.

**Comment by OPARA-PROSPER** [comment]
I'm a windows pro user. Though I'm a newbie to open source contribution but i'm willing to help, can i also help with the E2E test?

**Comment by ev1stensberg** [comment]
Sure. Have you written any before @OPARA-PROSPER ?
